ethereum/rpc/namespaces/miner/api.go

```diff
@@ -0,0 +1,163 @@
+package miner
+
+import (
+	"errors"
+
+	"github.com/cosmos/cosmos-sdk/client/flags"
+	codectypes "github.com/cosmos/cosmos-sdk/codec/types"
+	"github.com/cosmos/cosmos-sdk/server"
+	"github.com/cosmos/cosmos-sdk/server/config"
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	authtx "github.com/cosmos/cosmos-sdk/x/auth/tx"
+	"github.com/ethereum/go-ethereum/common"
+	"github.com/ethereum/go-ethereum/common/hexutil"
+	ethtypes "github.com/ethereum/go-ethereum/core/types"
+	"github.com/tendermint/tendermint/libs/log"
+	"github.com/tharsis/ethermint/ethereum/rpc/namespaces/eth"
+	rpctypes "github.com/tharsis/ethermint/ethereum/rpc/types"
+	evmtypes "github.com/tharsis/ethermint/x/evm/types"
+)
+
+// API is the miner prefixed set of APIs in the Miner JSON-RPC spec.
+type API struct {
+	ctx    *server.Context
+	logger log.Logger
+	ethAPI *eth.PublicAPI
+}
+
+// NewMinerAPI creates an instance of the Miner API.
+func NewMinerAPI(
+	ctx *server.Context,
+	ethAPI *eth.PublicAPI,
+) *API {
+	return &API{
+		ctx:    ctx,
+		ethAPI: ethAPI,
+		logger: ctx.Logger.With("module", "miner"),
+	}
+}
+
+// SetEtherbase sets the etherbase of the miner
+func (api *API) SetEtherbase(etherbase common.Address) bool {
+	api.logger.Debug("miner_setEtherbase")
+
+	list, err := api.ethAPI.ClientCtx().Keyring.List()
+	if err != nil || len(list) < 1 {
+		api.logger.Debug("Could not get list of addresses")
+		return false
+	}
+
+	addr := common.BytesToAddress(list[0].GetPubKey().Address())
+
+	api.logger.Info(addr.String())
+
+	args := rpctypes.SendTxArgs{}
+	args.From = addr
+	// TODO: set this as the message info
+	// delegatorAddress := addr
+	// withdrawAddress := etherbase
+	args.Data = &hexutil.Bytes{}
+	args, err = api.ethAPI.SetTxDefaults(args)
+	if err != nil {
+		api.logger.Debug("Unable to parse transaction args", "error", err.Error())
+		return false
+	}
+
+	msg := args.ToTransaction()
+
+	if err := msg.ValidateBasic(); err != nil {
+		api.logger.Debug("tx failed basic validation", "error", err.Error())
+		return false
+	}
+
+	signer := ethtypes.LatestSignerForChainID(args.ChainID.ToInt())
+
+	// Sign transaction
+	if err := msg.Sign(signer, api.ethAPI.ClientCtx().Keyring); err != nil {
+		api.logger.Debug("failed to sign tx", "error", err.Error())
+		return false
+	}
```

Comment:
you can sign using the `Sign` function from the SDK:

https://github.com/cosmos/cosmos-sdk/blob/v0.43.0-rc2/client/tx/tx.go#L387

you can get the `name` using the [`KeyByAddress`](https://github.com/cosmos/cosmos-sdk/blob/v0.43.0-rc2/crypto/keyring/keyring.go#L61) function from the keyring (you need to pass the delegator address that you obtained from the `backend.GetCoinbase` func)

Reply:
We implemented this changes